src/main/java/frc/trigon/robot/subsystems/collector/CollectorSubsystem.java

```diff
@@ -0,0 +1,41 @@
+package frc.trigon.robot.subsystems.collector;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.StartEndCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.trigon.robot.utilities.CurrentWatcher;
+import java.util.function.Supplier;
+
+public class CollectorSubsystem extends SubsystemBase {
+    private final static CollectorSubsystem INSTANCE = new CollectorSubsystem();
+    private final TalonSRX motor = CollectorConstants.MOTOR;
+    private final Supplier<Double> motorCurrent = CollectorConstants.MOTOR_CURRENT;
```

Comment:
Again, no need for this. You just just do motor::getSupplyCurrent in the CurrentWatcher's constructor.

Reply:
I tried making it in the constructor, it didn't work. even chatGPT didn't manage

Reply:
What exactly doesn't work? You just need to put motor::getSupplyCurrent in the creation of the current watcher

Reply:
no, but I need to make it a supplier. motor::getSupplyCurrent returns a double

Reply:
It doesn't.

Reply:
It returns a supplier of the current